scripts/zkai_d128_risc0_statement_receipt_gate.py

```
@@ -0,0 +1,618 @@
+def build_payload(
+    *,
+    prove: bool = False,
+    receipt_path: pathlib.Path = RECEIPT_OUT,
+    previous_proof_generation_time_ms: float | None = None,
+) -> dict[str, Any]:
+    journal = expected_journal_contract()
+    journal_bytes = expected_journal_bytes()
+    host_summary = generate_or_verify_receipt(prove=prove, receipt_path=receipt_path)
+    receipt_bytes = receipt_path.read_bytes()
+    if len(receipt_bytes) <= 0 or len(receipt_bytes) > MAX_RECEIPT_BYTES:
+        raise D128Risc0StatementReceiptError("receipt artifact size outside allowed bound", layer="receipt_artifact")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)